### PR TITLE
Some cleanup

### PR DESCRIPTION
**Pull request by arminasbrazenas**
Renamed the files, functions, did not change any logic. Changed printing to work only when in debug mode, since functions should not have side effects (as our lecturer said :)). If function fails, it returns false to indicate a failure.

### CODE REVIEW

File: lib/utils.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "utils.h"
```
Comment (AugustinasJarockis) [inline lib/utils.c:1]: Confusing name, maybe something like utilities.h would be better

File: lib/utils.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#ifndef UTILS_H
+#define UTILS_H
+
+#ifdef DEBUG
+# define DEBUG_PRINT(x) printf x
```
Comment (AugustinasJarockis) [inline lib/utils.h:5]: Negali pridėti skliaustų prie printf, kad nereiktų tų skliaustų papildomų visur rašinėti?

File: lib/utils.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+#else
+# define DEBUG_PRINT(x) do {} while (0)
```
Comment (AugustinasJarockis) [inline lib/utils.h:7]: Manau vietoj do {} while (0) papraščiau būtų tiesiog kokį ; ar {} padėti.

File: lib/utils.h (hunk 2 above)
Comment (AugustinasJarockis) [inline lib/utils.h:5]: Kodėl tarpas po #

File: lib/html-doubly-linked-list.h [hunk 4]
```diff
@@ -4,6 +4,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
+#include <stdbool.h>
```
Comment (AugustinasJarockis) [inline lib/html-doubly-linked-list.h:7]: Manau geriau naudoti int vietoje bool (kaip ir turėtum per C). Be to, naudojant int dar ir klaidos kodą galima gauti jei reikia

File: lib/html-doubly-linked-list.c [hunk 5]
```diff
@@ -1,16 +1,18 @@
 #include "html-doubly-linked-list.h"
+#include "utils.h"
```
Comment (AugustinasJarockis) [inline lib/html-doubly-linked-list.c:2]: Confusing name

File: lib/html-doubly-linked-list.c [hunk 6]
```diff
@@ -1,16 +1,18 @@
...
 
 static HTML *g_html_start = NULL;
```
Comment (AugustinasJarockis) [inline lib/html-doubly-linked-list.c:4]: static is absolutely useless here

File: lib/html-doubly-linked-list.c [hunk 7]
```diff
@@ -1,16 +1,18 @@
...
 static HTML *g_html_current = NULL;
```
Comment (AugustinasJarockis) [inline lib/html-doubly-linked-list.c:5]: Name looks absolutely hideous. Old one was better. It was immediately clear it was different

File: lib/main.c [hunk 8]
```diff
@@ -1,22 +1,22 @@
...
-    insert_tag(name2, NULL);
+    insert_html_tag(name2, NULL);
 
     printf("Tag inserted\n");
-    char *file_name = "output.txt";
+    char *file_name = "output.html";
     generate_html_file(file_name);
     printf("File printed\n");
-    delete_code();
+    cleanup();
```
Comment (AugustinasJarockis) [inline lib/main.c:21]: cleanup is very undescriptive

File: lib/html-doubly-linked-list.c [hunk 9]
```diff
@@ -1,16 +1,18 @@
...
-        if (file == NULL) {
-            printf("Failed to open file %s: %s\n", file_name, strerror(errno));
-            return;
+        if (file == NULL)
+        {
+            DEBUG_PRINT(("Failed to open file %s: %s\n", file_name, strerror(errno)));
+            return false;
         }
 
         HTML *html_current = g_html_start;
```
Comment (AugustinasJarockis) [inline lib/html-doubly-linked-list.c:18]: This name change is especially bad. It is VERY similar to g_html_current and I at first confused it with this variable. These two variables have nothing in common. Change this

**Comment by AugustinasJarockis** [comment]
Also, it was very difficult to follow what was changed because you changed the file names and everything was shown as changed.

File: lib/utils.h (hunk 2 above)
Comment (arminasbrazenas) [reply lib/utils.h]: fixed

File: lib/html-doubly-linked-list.c (hunk 7 above)
Comment (arminasbrazenas) [reply lib/html-doubly-linked-list.c:5]: no it's not, previously it was conflicting with standard c names, currently there is g prefix for global variable

File: lib/html-doubly-linked-list.h (hunk 4 above)
Comment (arminasbrazenas) [reply lib/html-doubly-linked-list.h:7]: true, maybe a lil later

File: lib/html-doubly-linked-list.c (hunk 5 above)
Comment (arminasbrazenas) [reply lib/html-doubly-linked-list.c]: changed

File: lib/html-doubly-linked-list.c (hunk 6 above)
Comment (arminasbrazenas) [reply lib/html-doubly-linked-list.c]: yes

**Pull request merged**